### PR TITLE
Add logger to cache entry to catch exception

### PR DESCRIPTION
**Pull request by lindexi**
As we can see, we can not know the EvictionCallback invoked failed. I want to add the logger to CacheEntry to log the exception

But I don't know if it is suitable to use LogInformation to log the exception

Thank you

### CODE REVIEW

**Comment by maryamariyan** [comment]
@lindexi there is no issue tracking this is there?

File: src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs [hunk 1]
```diff
@@ -318,10 +326,10 @@ private static void InvokeCallbacks(CacheEntry entry)
                 {
                     registration.EvictionCallback?.Invoke(entry.Key, entry.Value, entry.EvictionReason, registration.State);
                 }
-                catch (Exception)
+                catch (Exception e)
                 {
                     // This will be invoked on a background thread, don't let it throw.
-                    // TODO: LOG
+                    entry._logger.LogInformation(e, "EvictionCallback invoked failed");
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs:332]: FYI, the PR that introduced the //TODO: LOG comment: https://github.com/aspnet/Caching/pull/4/files

**Review by maryamariyan — APPROVED** [review]
LGTM, this could work. I retriggered CI to see what failures there are if any.

File: src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs [hunk 2]
```diff
@@ -19,6 +19,7 @@ internal class CacheEntry : ICacheEntry
         private IList<IDisposable> _expirationTokenRegistrations;
         private IList<PostEvictionCallbackRegistration> _postEvictionCallbacks;
         private bool _isExpired;
+        private readonly ILogger _logger;
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs:22]: missing namespace for ILogger is giving compile error

**Review by maryamariyan — CHANGES_REQUESTED** [review]
missing namespace on CacheEntry for ILogger is giving compile error

File: src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs (hunk 1 above)
Comment (Tratcher) [reply src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs:332]: It only took 6 years for someone to notice 😁.

Comment (Tratcher) [reply src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs:332]: This should probably be logged at the error level, it's application code that's throwing.

**Comment by lindexi** [comment]
@maryamariyan  Thank you and I do not know which the EventId is right to use.

File: src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs (hunk 1 above)
Comment (lindexi) [reply src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs]: @Tratcher Thank you and I replace the `LogInformation` to `LogError`

**Comment by maryamariyan** [comment]
close and reopening to retrigger the CI

**Comment by lindexi** [comment]
@maryamariyan  The CI is failed, and what should I do?

**Comment by maryamariyan** [comment]
The PR fails on the step to build native dependencies. @lindexi maybe because the PR was stale a little bit. Do you mind doing a rebase to update the PR?

**Review by maryamariyan — APPROVED** [review]

**Pull request merged**